src/backend/distributed/planner/insert_select_planner.c

```diff
@@ -878,14 +895,16 @@ Query *
 ReorderInsertSelectTargetLists(Query *originalQuery, RangeTblEntry *insertRte,
 							   RangeTblEntry *subqueryRte)
 {
+	Index insertTableId = originalQuery->resultRelation + 1;
```

Comment:
TODO: verify that this is always the case

Reply:
That `+ 1` thing doesn't make sense. But what is even more surprising is that setting insertTableId to 2 both makes the tests added in this pr passing and doesn't break any other tests, even if it was set to 1 previously.

Reply:
Seems that using incorrect varno for Insert Vars wasn't causing any problems before since no code-paths (other than combine query planner) needs to plan the Insert part of the query (hence needs to process its Vars)

Reply:
And the deparser already doesn't care about varnos when creating shard query since it only takes relation id of the table into the account